Modules/posixmodule.c

```diff
@@ -10064,6 +10069,252 @@ os_times_impl(PyObject *module)
 #endif /* HAVE_TIMES */
 
 
+#if defined(HAVE_TIMERFD_API)
+#define ONE_SECOND_IN_NS (1000 * 1000 * 1000)
+#define EXTRACT_NSEC(value)  (long)( ( (double)(value) - (time_t)(value) ) * 1e9)
+#define CONVERT_SEC_AND_NSEC_TO_DOUBLE(sec, nsec) ( (double)(sec) + (double)(nsec) * 1e-9 )
+
+static PyObject *
+build_itimerspec(const struct itimerspec* curr_value)
+{
+    double _next_expiration = CONVERT_SEC_AND_NSEC_TO_DOUBLE(curr_value->it_value.tv_sec,
+                                                          curr_value->it_value.tv_nsec);
+    PyObject *next_expiration = PyFloat_FromDouble(_next_expiration);
+    if (next_expiration == NULL) {
+        return NULL;
+    }
+    double _interval = CONVERT_SEC_AND_NSEC_TO_DOUBLE(curr_value->it_interval.tv_sec,
+                                                   curr_value->it_interval.tv_nsec);
+    PyObject *interval = PyFloat_FromDouble(_interval);
+    if (interval == NULL) {
+        Py_DECREF(next_expiration);
+        return NULL;
+    }
+    PyObject *tuple = PyTuple_Pack(2, next_expiration, interval);
+    Py_DECREF(interval);
+    Py_DECREF(next_expiration);
+    return tuple;
+}
+
+static PyObject *
+build_itimerspec_ns(const struct itimerspec* curr_value)
+{
+    long long _next_expiration = (long long)curr_value->it_value.tv_sec * ONE_SECOND_IN_NS +
+                    (long long)curr_value->it_value.tv_nsec;
+    PyObject *next_expiration = PyLong_FromLongLong(_next_expiration);
```

Comment:
`_PyTime_FromTimespec()` followed by `_PyTime_AsNanosecondsObject()` can be used here.

Reply:
@serhiy-storchaka 
changed to use _PyTime_FromTimespec